test/tests/functional/pbs_dup_acc_log_for_resv.py

```diff
@@ -47,16 +48,440 @@ class TestDupAccLogForResv(TestFunctional):
 
     def setUp(self):
         TestFunctional.setUp(self)
+        if len(self.moms) != 2:
+            self.skipTest('test requires two MoMs as input, ' +
+                          'use -p moms=<mom1>:<mom2>')
+
+    def check_log_resv_time(self, accounting_log_msg, start_tup):
+        """
+        This is a helper function to retrieve start and end time
+        of standing reservation from accounting log.
+        """
+        loglist_value = accounting_log_msg[start_tup][1]
+        log_value = loglist_value.split()
+        actual_reservation_time = []
+        check_status = ['start', 'end']
+        for status in check_status:
+            for attribute in log_value:
+                attribute_value = attribute.split("=")
+                if attribute_value[0] == status:
+                    actual_reservation_time.append(int(attribute_value[1]))
+
+        return actual_reservation_time
+
+    def differentiate_resv_instance(self, expected_start, expected_end,
+                                    got_start, got_end, reserve_index):
+        """
+        This is a helper function to differentiate log match for each
+        instance(reserve_index) of standing reservation.
+        As input function accept expected_start,expected_end,got _start,
+        got_end time value and reserve_ index.
+        By comparison of expected_start & got_start and expected_end &
+        got_end time value,function will log info for specific
+        instance(reserve_index) of standing reservation.
+        """
+
+        self.assertTrue(((expected_start == got_start) and
+                         (expected_end == got_end)),
+                        'Time in the accounting log does not match with'
+                        'the expected time for the standing reservation')
+
+        match_string = 'Matched accounting log for instance' + reserve_index
+        if reserve_index in ('1', '2', '3'):
+            self.logger.info(match_string)
+        else:
+            self.assertTrue(False, "Unknown Instance")
 
     def test_accounting_logs(self):
+        """
+        Test for duplicate records in accounting log for advance reservations
+        on restart of server
+        """
         r1 = Reservation(TEST_USER)
         a = {'Resource_List.select': '1:ncpus=1', 'reserve_start': int(
             time.time() + 5), 'reserve_end': int(time.time() + 60)}
         r1.set_attributes(a)
         r1id = self.server.submit(r1)
-        time.sleep(8)
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=r1id, max_attempts=30, offset=8)
+
         self.server.restart()
         m = self.server.accounting_match(
```

Comment:
As per latest update in log_match now yo don't need to check return value of log_match so please remove assertNotEqual.
This applies to whole file.